src/types/redis_json.cc

```diff
@@ -117,4 +117,30 @@ rocksdb::Status Json::Get(const std::string &user_key, const std::vector<std::st
   return rocksdb::Status::OK();
 }
 
+rocksdb::Status Json::StrAppend(const std::string &user_key, const std::vector<std::string> &paths, const std::string &value, uint64_t &deleted_cnt) {
+  deleted_cnt = 0;
+  auto ns_key = AppendNamespacePrefix(user_key);
+
+  LockGuard guard(storage_->GetLockManager(), ns_key);
+
+  std::string bytes;
+  JsonMetadata metadata;
+  Slice rest;
+  auto s = GetMetadata(kRedisJson, ns_key, &bytes, &metadata, &rest);
+  if (!s.ok()) return s;
+
+  if (metadata.format != JsonStorageFormat::JSON)
+    return rocksdb::Status::NotSupported("JSON storage format not supported");
+
+  auto json_res = JsonValue::FromString(rest.ToStringView());
+  if (!json_res) return rocksdb::Status::Corruption(json_res.Msg());
+  auto json_val = *std::move(json_res);
+
+  for (const auto &path : paths) {
+    auto append_res = json_val.Append(path, value, deleted_cnt);
+    if (!append_res) return rocksdb::Status::InvalidArgument(append_res.Msg());
+  }
+  return write(ns_key, &metadata, json_val);
```

Comment:
Can we avoid a `write` when nothing changed in underlying json (.i.e `deleted_count == 0` in your case )